src/AutoRest.CSharp/Common/Output/PostProcessing/ReferenceMapBuilder.cs

```diff
@@ -127,11 +127,22 @@ private async Task ProcessExtensionSymbol(INamedTypeSymbol extensionClassSymbol,
                 if (!methodSymbol.IsExtensionMethod)
                     continue;
 
-                // find which document is using this extension method, and add it to the map
-                foreach (var reference in await SymbolFinder.FindReferencesAsync(methodSymbol, _project.Solution))
-                {
-                    await AddReferenceToReferenceMapAsync(extensionClassSymbol, reference, referenceMap, documentCache);
-                }
+                // this is to hook the extension class like this:
+                // internal static class FooExtensions
+                // {
+                //     public static string ToSerialString(this Foo foo) => foo.ToString();
+                //     public static Foo ToFoo(this string foo) => // omit body
+                // }
+
+                // if this is an extension method, we add it to the reference map of the type it is extending to pretend that this class is a part of that type
+                // handle the first method above
+                if (methodSymbol.Parameters[0].Type is INamedTypeSymbol typeSymbol)
```

Comment:
Is that possible that the method has no parameter at all?

Reply:
make sense. Thanks for pointing out! Let me fix this

Reply:
well after a thought - we have a check whether this method is extension method above, and we only get here when it is an extension method, therefore it is guaranteed to have at least one parameter.
But nevertheless I still made the change